provisionersdk/scripts/bootstrap_linux.sh

```diff
@@ -86,4 +86,12 @@ fi
 
 export CODER_AGENT_AUTH="${AUTH_TYPE}"
 export CODER_AGENT_URL="${ACCESS_URL}"
-exec ./$BINARY_NAME agent
+
+output=$(./${BINARY_NAME} --version | head -n1)
+if ! echo "${output}" | grep -q Coder; then
+	echo >&2 "ERROR: Downloaded agent binary is invalid"
+	echo >&2 "Script output: '${output}'"
+	exit 2
+fi
```

Comment:
I think we can be a bit more specific in what diagnostic we are running:

```suggestion
if ! echo "${output}" | grep -q Coder; then
	echo >&2 "ERROR: Downloaded agent binary returned unexpected version output"
	echo >&2 "${BINARY_NAME} --version output: '${output}'"
	exit 2
fi
```

Reply:
Fair 👍